app/uk/gov/hmrc/customs/rosmfrontend/controllers/HasExistingEoriController.scala

```diff
@@ -41,6 +45,20 @@ class HasExistingEoriController @Inject()(
         Future.successful(Ok(hasExistingEoriView(service, existingEori.id)))
   }
 
+  def enrol(service: Service.Value): Action[AnyContent] =
+    ggAuthorisedUserWithEnrolmentsAction { implicit request => user: LoggedInUserWithEnrolments =>
+      enrolmentService.enrolWithExistingCDSEnrolment(user, service.toString.toLowerCase).map {
+        case NO_CONTENT => Redirect(routes.HasExistingEoriController.enrolSuccess(service))
+        case _ => BadRequest("Enrolment failed")
```

Comment:
Do we have a design for failure?

Reply:
Good question.  I presumed this would result in the normal "something has gone wrong" page.  The one I got before you fixed the stub.   How is your success/failure stub changes going?

Reply:
Stying in the same place, doing different things :(

so here probably we want to throw some exception if we're not handling this with a specific page. Worth to check with design

Reply:
But this can be done outside of this PR